### ISSUE TITLE
Unicode characters are transformed upon rewrite

### ISSUE THREAD

**Issue by mbvyd** [xoofx/Tomlyn#38]
If a unicode character is in string value in file, then, upon file rewrite (that value is not changed in model) it is replaced with actual characters.
For example, was:
`key = "\U00002139 some text \U0001F4DD some text \U00002B07"`
After file update
```csharp 
await WriteModel(model);
```
it changes to:
`key = "ℹ some text 📝 some text ⬇"`
Afterwards this string value is displayed as it should, but I wonder is it possible to preserve unicode characters in file in "initial" format? Or it is by default and something is wrong with my setup?

I use custom model and following "wrap" (mentioned in doc) for every class in custom model (if that matters):
```csharp
internal class Entity : ITomlMetadataProvider
{
    // class contents before
    
    // storage for comments and whitespace
    TomlPropertiesMetadata? ITomlMetadataProvider.PropertiesMetadata { get; set; }

    // class contents after
}
```

PS: thanks for awesome library!

**Comment by xoofx** [comment]
> Afterwards this string value is displayed as it should, but I wonder is it possible to preserve unicode characters in file in "initial" format? 

Indeed, the string is loaded and converted to a runtime format for both the Syntax model and the runtime model. Supporting full roundtrip for string is probably possible but quite annoying. You need to keep an index to the characters transformed with the original format, make sure that it is the same string and attach some information on the side to rewrite the string when going back to TOML, or e.g keep the original raw string around with the original transformed string, if the original transformed string is the same, you can keep the original raw string.

Personally, I don't have a need for such a use case, but PR welcome if you find a solution that is not too disruptive. 🙂

**Comment by markusschaber** [comment]
Do you need to exactly match the input? Or is it just about escaping certain symbols, as your channel is not fully UTF-8 safe?

Maybe it would make sense to add a "global" setting to just use the escaped form for certain character classes when writing strings, e. g. all non-BML codepoints, or all non-ASCII characters.
